src/NuGet.Protocol.Core.v3/HttpHandlerResourceV3Provider.cs

```diff
@@ -104,6 +104,20 @@ private class ProxyWebRequestHandler : WebRequestHandler
                             HttpHandlerResourceV3.ProxyPassed(Proxy);
                         }
 
+                        if (response.StatusCode == HttpStatusCode.Unauthorized)
+                        {
+                            if (HttpHandlerResourceV3.PromptForProxyCredentials == null)
+                            {
+                                // Fail this operation.
+                                response.EnsureSuccessStatusCode();
+                            }
+                            else
+                            {
+                                // PromptForProxyCredentials is a misnomer. It can prompt for proxy credentials or request credentials.
+                                Credentials = HttpHandlerResourceV3.PromptForProxyCredentials(request.RequestUri, /* webProxy */ null);
```

Comment:
It looks like your change makes v3 resource works with authenticated feeds, but removes the capability of working thru authenticated proxies.